### PR TITLE
Windows fixes (make test suite pass on windows)

### PR DESCRIPTION
**Pull request by dalito**
This fixes all windows issues and adds Windows to the CI test matrix.

To pass all tests without error this PR requires the fixes in https://github.com/INCATools/ontology-access-kit/pull/345

### CODE REVIEW

File: .github/workflows/check-pull-request.yaml [hunk 1]
```diff
@@ -51,26 +63,26 @@ jobs:
...
+
 #       - name: Run image
 #         uses: abatilo/actions-poetry@v2.0.0
 #         # with:
 #         #   poetry-version: ${{ matrix.poetry-version }}
 #       - name: View poetry --help
 #         run: poetry --help
 
-      # Make test
-      - name: Make test
```
Comment (sierra-moxon) [inline .github/workflows/check-pull-request.yaml:74]: Thanks @dalito for working on this for windows! :) - is this change for consistency, or does `make test` not work for windows?

Comment (dalito) [reply .github/workflows/check-pull-request.yaml:74]: make is not available on windows.

**Comment by dalito** [comment]
The failure is due to pinning the very old scipy version (1.6.1). The poetry lock file must be updated. I was not sure if I should commit the changed file but will do so in a minute.

**Comment by dalito** [comment]
This should start to pass the tests after the next release of oaklib (v0.1.53) and after updating poetry.lock.

**Comment by dalito** [comment]
FYI: So far I have no idea what causes the failiure. With exactly the same package versions as in CI, I get no error when I run the test locally on Win10. But I'll continue investigating.

**Comment by dalito** [comment]
The error is caused by the line endings created by git checkout. Locally I work with the unix convention (LF). However, the windows pipeline uses the windows standard which is CRLF. If I convert my local file `tests/resources/prof.ofn` from LF to CRLF line endings I get the same error as in the CI pipeline. So the root of the problem is that the text (yaml) file is opened in binary mode. This happens in funowl, https://github.com/linkml/schema-automator/actions/runs/3415094925/jobs/5683997986#step:8:60

**Comment by dalito** [comment]
@sierra-moxon - Can you restart the workflow? As a first time contributor in this repo I can't do it myself.

Using .gitattributes to set the line-ending to LF should decouple this PR from waiting for a fix in funowl.

**Pull request merged**